paper-server/src/main/java/org/bukkit/craftbukkit/block/CraftBiome.java

```diff
@@ -40,8 +46,30 @@ public static Biome minecraftHolderToBukkit(Holder<net.minecraft.world.level.bio
         return CraftRegistry.bukkitToMinecraftHolder(bukkit, Registries.BIOME);
     }
 
+    private final PaperBiomeClimate climate;
+    private final PaperBiomeSpecialEffects effects;
+    private final PaperBiomeMobSpawning mobSpawning;
+
     public CraftBiome(final Holder<net.minecraft.world.level.biome.Biome> holder) {
         super(holder, count++);
+        this.climate = new PaperBiomeClimate(holder);
+        this.effects = new PaperBiomeSpecialEffects(holder);
+        this.mobSpawning = new PaperBiomeMobSpawning(holder);
+    }
+
+    @Override
+    public @NotNull BiomeClimate climate() {
```

Comment:
This class is already NullMarked, so no need to add those annotations individually